returnn/tf/layers/basic.py

```diff
@@ -5776,7 +5777,15 @@ def get_out_data_from_opts(cls, n_out=NotSpecified, out_type=None, sources=(), *
       if isinstance(out_type, dict):
         out_type_.update(out_type)
       elif callable(out_type):
-        out_type_ = out_type  # just overwrite
+        def call_out_type_with_eval_locals(**out_type_kwargs):
+          """
+          :param out_type_kwargs:
+          :rtype: Data
+          """
+          out_type_kwargs = out_type_kwargs.copy()
+          out_type_kwargs.update(eval_locals or {})
+          return out_type(out_type_kwargs)
```

Comment:
```suggestion
          return out_type(**out_type_kwargs)
```
?
(Did you test this?)

Reply:
Sorry, my original code did not run in Python 2, and I did not test the change I pushed just now :sweat_smile: